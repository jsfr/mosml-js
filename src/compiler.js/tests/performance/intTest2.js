"use strict"
var ctimeBegin$1 = function($_var1)
{return console.time($_var1);};
var ctimeEnd$2 = function($_var5)
{return console.timeEnd($_var5);};
var repeat$3 = function($_var9)
{return function($_var10)
{return function($_var11){
var $_var13 = function($_var14)
{return (function(){switch($_var14){
case 0:
return $_var10($_var11)
default:
return ($_var10($_var11), $_var13($_mosmllib.overflowCheck64($_var14-1)))
}}());};
return $_var13($_mosmllib.overflowCheck64($_var9-1));};};};
var timerep$4 = function($_var35)
{return function($_var36)
{return function($_var37)
{return repeat$3($_var35)(function($_var41)
{return (ctimeBegin$1($_var37), ($_var36($_var41), ctimeEnd$2($_var37)));});};};};
var recAdd$5 = function($_var54){
var $_var56 = $_mosmllib.Constructor(250,[0.0]);
return (function(){while ($_mosmllib.Constructor($_var56.args[0] < $_var54? 1 : 0).tag){
((function(){$_var56.args[0] = $_var56.args[0]+1.0}()), $_mosmllib.overflowCheck64(10+10))
}}());};
var recSub$6 = function($_var73){
var $_var75 = $_mosmllib.Constructor(250,[0.0]);
return (function(){while ($_mosmllib.Constructor($_var75.args[0] < $_var73? 1 : 0).tag){
((function(){$_var75.args[0] = $_var75.args[0]+1.0}()), $_mosmllib.overflowCheck64(10-1))
}}());};
var it$7 = timerep$4(10)(recAdd$5)("addInt")(10000000.0);
var it$8 = timerep$4(10)(recAdd$5)("subInt")(10000000.0);
