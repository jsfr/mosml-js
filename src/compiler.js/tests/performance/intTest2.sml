prim_val ctimebegin_ : string -> unit = 1 "console.time";
prim_val ctimeend_ : string -> unit = 1 "console.timeEnd";
fun ctimeBegin n = ctimebegin_ n;
fun ctimeEnd n = ctimeend_ n;
fun repeat n f x =
    let val n = ref n
    in  while !n > 0 do (n := !n-1; f x) end;
fun timerep n f s = repeat n (fn x => (ctimeBegin(s);f x; ctimeEnd(s)));

fun intAdd x =
  let
    val addRef = ref 0.0;
  in
    while !addRef < x do (addRef := !addRef+1.0; 10+10)
  end

fun intSub x =
  let
    val subRef = ref 0.0;
  in
    while !subRef < x do (subRef := !subRef+1.0; 10-1)
  end


val it = timerep 100 intAdd "intAdd" 1000000.0
val it = timerep 100 intSub "intSub" 1000000.0




