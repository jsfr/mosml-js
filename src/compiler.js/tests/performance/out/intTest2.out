beautified intTest2.js
==============================JS BEGIN==============================
addInt2: 88ms
addInt2: 87ms
addInt2: 87ms
addInt2: 84ms
addInt2: 85ms
addInt2: 85ms
addInt2: 86ms
addInt2: 84ms
addInt2: 85ms
addInt2: 85ms
addInt2: 86ms
addInt2: 92ms
addInt2: 92ms
addInt2: 92ms
addInt2: 110ms
addInt2: 99ms
addInt2: 99ms
addInt2: 91ms
addInt2: 92ms
addInt2: 92ms
addInt2: 93ms
addInt2: 93ms
addInt2: 93ms
addInt2: 92ms
addInt2: 92ms
addInt2: 93ms
addInt2: 92ms
addInt2: 93ms
addInt2: 94ms
addInt2: 92ms
addInt2: 92ms
addInt2: 93ms
addInt2: 92ms
addInt2: 92ms
addInt2: 93ms
addInt2: 101ms
addInt2: 92ms
addInt2: 91ms
addInt2: 90ms
addInt2: 86ms
addInt2: 84ms
addInt2: 86ms
addInt2: 87ms
addInt2: 87ms
addInt2: 84ms
addInt2: 85ms
addInt2: 84ms
addInt2: 93ms
addInt2: 92ms
addInt2: 92ms
addInt2: 93ms
addInt2: 92ms
addInt2: 92ms
addInt2: 93ms
addInt2: 92ms
addInt2: 92ms
addInt2: 92ms
addInt2: 93ms
addInt2: 107ms
addInt2: 98ms
addInt2: 97ms
addInt2: 91ms
addInt2: 92ms
addInt2: 94ms
addInt2: 92ms
addInt2: 92ms
addInt2: 92ms
addInt2: 92ms
addInt2: 95ms
addInt2: 92ms
addInt2: 90ms
addInt2: 85ms
addInt2: 88ms
addInt2: 85ms
addInt2: 86ms
addInt2: 85ms
addInt2: 87ms
addInt2: 85ms
addInt2: 86ms
addInt2: 87ms
addInt2: 157ms
addInt2: 109ms
addInt2: 96ms
addInt2: 92ms
addInt2: 93ms
addInt2: 92ms
addInt2: 93ms
addInt2: 92ms
addInt2: 92ms
addInt2: 92ms
addInt2: 92ms
addInt2: 93ms
addInt2: 93ms
addInt2: 92ms
addInt2: 92ms
addInt2: 93ms
addInt2: 93ms
addInt2: 92ms
addInt2: 93ms
addInt2: 92ms
subInt2: 112ms
subInt2: 109ms
subInt2: 100ms
subInt2: 92ms
subInt2: 97ms
subInt2: 92ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 92ms
subInt2: 93ms
subInt2: 102ms
subInt2: 99ms
subInt2: 101ms
subInt2: 93ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 89ms
subInt2: 84ms
subInt2: 92ms
subInt2: 87ms
subInt2: 86ms
subInt2: 85ms
subInt2: 87ms
subInt2: 84ms
subInt2: 86ms
subInt2: 85ms
subInt2: 84ms
subInt2: 93ms
subInt2: 92ms
subInt2: 92ms
subInt2: 91ms
subInt2: 94ms
subInt2: 109ms
subInt2: 99ms
subInt2: 96ms
subInt2: 93ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 92ms
subInt2: 92ms
subInt2: 92ms
subInt2: 92ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 92ms
subInt2: 92ms
subInt2: 92ms
subInt2: 91ms
subInt2: 92ms
subInt2: 146ms
subInt2: 172ms
subInt2: 101ms
subInt2: 93ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 92ms
subInt2: 92ms
subInt2: 92ms
subInt2: 92ms
subInt2: 93ms
subInt2: 92ms
subInt2: 92ms
subInt2: 104ms
subInt2: 99ms
subInt2: 98ms
subInt2: 92ms
subInt2: 92ms
subInt2: 94ms
subInt2: 92ms
subInt2: 91ms
subInt2: 92ms
subInt2: 92ms
subInt2: 92ms
subInt2: 92ms
subInt2: 93ms
subInt2: 98ms
subInt2: 93ms
==============================JS END================================
